rest_authenticate/services/session.py

```diff
@@ -0,0 +1,55 @@
+import datetime
+
+from odoo import fields
+from odoo.http import request, root
+from odoo.service import security
+
+from odoo.addons.base_rest import restapi
+from odoo.addons.component.core import Component
+
+
+def _rotate_session(httprequest):
+    if httprequest.session.rotate:
+        root.session_store.delete(httprequest.session)
+        httprequest.session.sid = root.session_store.generate_key()
+        if httprequest.session.uid:
+            httprequest.session.session_token = security.compute_session_token(
+                httprequest.session, request.env
+            )
+        httprequest.session.modified = True
+
+
+class SessionAuthenticateService(Component):
+    _inherit = "base.rest.service"
+    _name = "session.authenticate.service"
+    _usage = "authenticate"
+    _collection = "session.rest.services"
+
+    @restapi.method([(["/"], "POST")])
+    def authenticate(self):
+        params = request.params
+        request.session.authenticate(params["db"], params["login"], params["password"])
+        result = request.env["ir.http"].session_info()
+        if request.session.rotate:
+            # avoid to rotate the session outside of the scope of this method
+            # to ensure that the session ID does not change after this method
+            _rotate_session(request)
+            request.session.rotate = False
+        expiration = datetime.datetime.utcnow() + datetime.timedelta(days=90)
+        result["session"] = {
+            "sid": request.session.sid,
+            "expires_at": fields.Datetime.to_string(expiration),
+        }
+        return result
+
+
+class SessionLogoutService(Component):
+    _inherit = "base.rest.service"
+    _name = "session.logout.service"
+    _usage = "logout"
+    _collection = "session.rest.services"
```

Comment:
maybe a base component to inherit from?

Reply:
Probably; I first thought of doing everything in the same class, but that means that I will end up with a uselessly long url: `[controller route]/[_usage]/[service route]`, something like `session_rest/session/authenticate` in my case. Otherwise I can do it with the "old style" API, but I figured "new is always better" :-)

Reply:
I'd change the root path to `/session/` and have only `_usage = "auth"` so that you'll have something like `/session/auth/login|logout`

Reply:
Ok good idea.

Reply:
:+1: for @simahawk proposal